detection-rules/attachment_pdf_extortion_leveraging_breach_data.yml

```diff
@@ -21,7 +21,9 @@ source: |
   // body contains a US address, or the body is the subject
   and (
     regex.icontains(body.current_thread.text,
-                    '\d+\s[\w\s.]+(?:\n)?[\w\s]+\s[A-Z]{2}\s\d{5}(?:-\d{4})?(?:\n)?'
+                    '\d+\s[\w\s.]+(?:\n)?[\w\s]+\s[A-Z]{2}\s\d{5}(?:-\d{4})?(?:\n)?',
+                    // a Canadian address
+                    '\d+\s[\w\s.]+(?:\n)?[\w\s]+\s((?:Ontario|ON)|(?:Quebec|QC)|(?:Nova Scotia|NS)|(?:New Brunswick|NB)|(?:Manitoba|MB)|(?:British Columbia|BC)|(?:Prince Edward Island|PE)|(?:Saskatchewan|SK)|(?:Alberta|AB)|(?:Newfoundland and Labrador|NL))\s*[ABCEGHJ-NPRSTVXY]\d[ABCEGHJ-NPRSTV-Z][ -]?\d[ABCEGHJ-NPRSTV-Z]\d'
```

Comment:
`(?:Prince Edward Island|PE)`

You many want to add `PEI` as well, it's a common abbreviation for this province.

---

Also missing is:
- `Yukon|YT`
- `Northwest Territories|NT`
- `Nunavut|NU`